tools/serve/serve.py

```diff
@@ -73,7 +73,9 @@ def __call__(self, request, response):
         self.handler(request, response)
 
     def handle_request(self, request, response):
-        for header_name, header_value in self.headers:
+        headers = self.headers + handlers.load_headers(
```

Comment:
Optional: I'd consider renaming `self.headers` to something like `self.base_headers`. It's just a bit weird to say "these are the headers, but then we have to add more headers on-top to make up the headers".

Reply:
This is pretty much an implementation detail (despite being public), as it's not intended/suitable for direct use, so I'm not very worried about the name here.